pkgs/top-level/all-packages.nix

```diff
@@ -25959,6 +25959,8 @@ with pkgs;
   aucatctl = callPackage ../applications/audio/aucatctl { };
 
   audacious = libsForQt5.callPackage ../applications/audio/audacious { };
+  audacious-bare = libsForQt5.callPackage ../applications/audio/audacious/player.nix { };
+  audacious-plugins = libsForQt5.callPackage ../applications/audio/audacious/plugins.nix { };
```

Comment:
I am exposing those derivations for convenience, but we can also not chose to not expose them. Specially `audacious-bare` is kinda useless right now since it fails with:

```
ERROR plugin-init.cc:155 [start_required]: No output plugin found.
(Did you forget to install audacious-plugins?)
zsh: abort (core dumped)  ./result/bin/audacious
```

Reply:
However, exposing the `audacious-bare` will be useful for the folks that want to run customized `audacious-plugins`, for example.

Reply:
Decided to do one more refactor and add a `withPlugins` flag instead to `audacious`. This made the code more clear.